dockerfiles/init/modules/openshift/files/scripts/ocp.sh

```diff
@@ -77,6 +81,21 @@ export CONFIG_DIR=${CONFIG_DIR:-${DEFAULT_CONFIG_DIR}}
 
 }
 
+test_dns_provider() {
+    #add current $DNS_PROVIDER to the providers list to respect environment settings
+    DNS_PROVIDERS=("$DNS_PROVIDER" "${DNS_PROVIDERS[@]}")
+    for i in ${DNS_PROVIDERS[@]}
+        do
+        if [[ $(dig +short 10.0.0.1.$i) = "10.0.0.1" ]]; then
```

Comment:
should we use dig or only "host" command ?
also should it have some timeout ? if one of DNS provider is not replying we might have delays

Reply:
@benoitf
1)  I feel more comfortable with `dig` host output should be parsed to get the IP while dig allows to get only IP
2) The reason why I didn't add timeout is because `dig` has own very short timeout.
real host resolve time:
```
time dig +short 10.0.0.1.nip.io
10.0.0.1

real	0m0.010s
user	0m0.002s
sys	0m0.003s
```

fake host resolve time:
```
time dig +short thisisfakehostnamepiupiu.com

real	0m0.055s
user	0m0.003s
sys	0m0.004s
```

Reply:
ok for dig.

By timeout, I was talking about when DNS provider is not responding/timeouting, not with an invalid domain name as in that case there is a DNS that reply that it is not existing.

Reply:
+1 to what @benoitf said. In your example DNS is resolved fast but IP is not responding whereas the problem might be in DNS